subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -18,6 +18,13 @@ static struct k_thread ztest_thread;
 #endif
 static bool failed_expectation;
 
+#ifdef CONFIG_SHELL
```

Comment:
Instead of automatically enabling this when the shell is enabled, should we introduce a separate config (e.g. `CONFIG_ZTEST_SHELL`) to control this feature so that we do not unnecessarily increase the footprint of the existing shell tests?

Reply:
right, this will be needed anyways, otherwise any test enabling the shell as we have it now, will end up being stuck waiting for input.